File: src/main/java/me/nepnep/nepbot/message/command/commands/Shutdown.java
package me.nepnep.nepbot.message.command.commands;

import me.nepnep.nepbot.Main;
import me.nepnep.nepbot.message.command.Category;
import me.nepnep.nepbot.message.command.ICommand;
import net.dv8tion.jda.api.Permission;
import net.dv8tion.jda.api.entities.TextChannel;
import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent;

import java.util.List;

public class Shutdown implements ICommand {
    @Override
    public void execute(List<String> args, GuildMessageReceivedEvent event) {
        long operator = Main.config.get("operator").longValue();
        TextChannel channel = event.getChannel();
        if (event.getAuthor().getIdLong() == operator) {
            channel.sendMessage("Shutting down").complete();
            event.getJDA().shutdown();
            System.exit(0);
        } else {
            channel.sendMessage("You do not have permission to use this command").queue();
        }
    }

    @Override
    public String getInvoke() {
        return "shutdown";
    }

    @Override
    public Permission requiredPermission() {
        return null;
    }

    @Override
    public Category getCategory() {
        return Category.GENERAL;
    }

    @Override
    public String getDescription() {
        return "Shuts down the bot";
    }
}

File: src/main/kotlin/me/nepnep/nepbot/message/command/commands/Shutdown.kt
package me.nepnep.nepbot.message.command.commands

import me.nepnep.nepbot.config
import me.nepnep.nepbot.message.command.Category
import me.nepnep.nepbot.message.command.AbstractCommand
import net.dv8tion.jda.api.entities.GuildMessageChannel
import net.dv8tion.jda.api.events.message.MessageReceivedEvent
import kotlin.system.exitProcess

class Shutdown : AbstractCommand(
    "shutdown",
    Category.GENERAL,
    "Shuts down the bot"
) {
    override suspend fun execute(args: List<String>, event: MessageReceivedEvent, channel: GuildMessageChannel) {
        val operator = config["operator"].longValue()

        if (event.author.idLong != operator) {
            channel.sendMessage("You do not have permission to use this command").queue()
            return
        }
        channel.sendMessage("Shutting down").complete()
        exitProcess(0)
    }
}